src/core/InstrumentPlayHandle.cpp

```diff
@@ -37,5 +40,41 @@ InstrumentPlayHandle::InstrumentPlayHandle( Instrument * instrument, InstrumentT
 	setAudioPort( instrumentTrack->audioPort() );
 }
 
+void InstrumentPlayHandle::play( sampleFrame * _working_buffer )
+{
+	InstrumentTrack * instrumentTrack = m_instrument->instrumentTrack();
+
+	// ensure that all our nph's have been processed first
+	ConstNotePlayHandleList nphv = NotePlayHandle::nphsOfInstrumentTrack(instrumentTrack, true );
+	
+	bool nphsLeft;
+	do
+	{
+		nphsLeft = false;
+		for( const NotePlayHandle * constNotePlayHandle : nphv )
+		{
+			NotePlayHandle * notePlayHandle = const_cast<NotePlayHandle *>( constNotePlayHandle );
```

Comment:
Out of scope of this PR. With commit 73f9f36c9af I have moved the cast closer to the `process` method because that method is the reason that the cast is needed in the first place. The other checks work with the const instances.

In general it seems to be a problem of unclear responsibilities.